docs/getting-started/concepts/dataset.md

```diff
@@ -0,0 +1,43 @@
+# Dataset
+
+Feast datasets allow for conveniently saving dataframes that include both features and entities to be subsequently used for data analysis and model training.
+[Data Quality Monitoring](https://docs.google.com/document/d/110F72d4NTv80p35wDSONxhhPBqWRwbZXG4f9mNEMd98) was the primary motivation for creating dataset concept.
+
+Dataset's metadata is stored in the Feast registry and raw data (features, entities, additional input keys and timestamp) is stored in the [offline store](../architecture-and-components/offline-store.md).
+
+Dataset can be created from:
+1. Results of historical retrieval
+2. [planned] Logging request (including input for [on demand transformation](../../reference/alpha-on-demand-feature-view.md)) and response during feature serving
+3. [planned] Logging features during writing to online store (from batch source or stream)
+
+
+### Creating Saved Dataset from Historical Retrieval
+
+To create a saved dataset from historical features for later retrieval or analysis, use the `save_as` parameter of `get_historical_features` method.
+If this parameter is supplied, Feast eagerly computes the dataframe and stores the data using specified `storage`.
+Any offline store can be used as such storage. Out of the box Feast supports BigQuery, Redshift and File storage (local, Google Storage or S3).
+
+```python
+
+from feast import FeatureStore
+from feast.saved_dataset import SavedDatasetOptions
+from feast.infra.offline_stores.bigquery_source import SavedDatasetBigQueryStorage
+
+store = FeatureStore()
+
+store.get_historical_features(
```

Comment:
Yep, this could work. But what are benefits of this approach? 
1. Do you want to reuse `offline_job`? But in that case, `historical` query can be run several times, which is additional costs. Whereas in the current proposed version saved dataset plays the role of cache. Eg,
```
result = store.get_historical_features(..., save_as='...')
result.to_df()
result.to_df() # both times reads from persisted table, not running the original query
```
2. Will `create_saved_dataset` run `offline_job` eagerly & what is the purpose of `SavedDatasetJob`?

Reply:
> But what are benefits of this approach?

Mostly consistency: `store.get_historical_features()` always return an `OfflineRetrievalJob` object

> Do you want to reuse offline_job?

I don't want to have strong assumptions on that. What I know is that the BQ Retrieval job has a `dry-run` option. So I would not be surprised if some users use it multiple times.

> query can be run several times, which is additional costs

That's on the user to decide if this additional cost is useful or not IMO

> Whereas in the current proposed version saved dataset plays the role of cache

I get that and this is great 👍 

> what is the purpose of SavedDatasetJob?

Now I understand it better, I think that it's not needed. Here is my updated example

```python
store = FeatureStore()
offline_job: OfflineRetrievalJob = store.get_historical_features(
    features=["driver:avg_trip"],
    entity_df=...,
)

retrieval_job: OfflineRetrievalJob = store.create_saved_dataset( # That operation executes the RetrievalJob
    name='my_training_dataset',
    storage=SavedDatasetBigQueryStorage(table_ref='<gcp-project>.<gcp-dataset>.my_training_dataset'),
    retrieval_job=offline_job,
    tags={'author': 'oleksii'},
)

retrieval_job.to_df()
```

As I said, I am thinking out-loud

Reply:
API is updated.